src/control/cmd/daos_agent/security_rpc.go

```diff
@@ -33,32 +53,106 @@ type (
 	SecurityModule struct {
 		log            logging.Logger
 		signCredential credSignerFn
+		credCache      *credentialCache
 
 		config *securityConfig
 	}
 )
 
-// NewSecurityModule creates a new module with the given initialized TransportConfig
+// NewSecurityModule creates a new module with the given initialized TransportConfig.
 func NewSecurityModule(log logging.Logger, cfg *securityConfig) *SecurityModule {
+	var credCache *credentialCache
+	credSigner := auth.GetSignedCredential
+	if cfg.credentials.CacheLifetime > 0 {
+		cache := &credentialCache{
```

Comment:
This should be `credCache`, right?

Reply:
Oops... Missed this in a refactoring. Good catch. Updated the unit test to use the constructor, as that would have caught this.